examples/sdk_examples.rb

```diff
@@ -140,3 +140,19 @@ def create(params)
   trace = AppOpticsAPM::SDK.current_trace
   AppOpticsAPM.logger.warn "Some log message #{trace.for_log}"
 end
+
+###############################################################
+# START A TRACE AND PROFILE
+###############################################################
+#
+# AppOpticsAPM::Profiling.run
+# This method adds profiling for the code executed in the block
+
+AppOpticsAPM::SDK.start_trace("#{name}_profiling") do
+  AppOpticsAPM::Profiling.run do
```

Comment:
When would a customer use `AppOpticsAPM::Profiling.run` and what would happen if the request is already being profiled and then `AppOpticsAPM::Profiling.run` is called?

Reply:
This q reminded me that I have to create a proper SDK function for profiling: ticket created: AO-20371

If it is already running, nothing happens
Profiling registers the thread IDs it profiles, if it is already registered profiling will continue as is, it will not be restarted or profiled twice

Reply:
actually, I removed that check, because there is a chance when something goes wrong that the data structure shows a thread being profiled when it is not
Profiling will restart and write a new entry event, I guess that will render the profiling events in the collector invalid
I'll make sure that the SDK guards against calling profiling twice

Reply:
Is the ProfileSpans KV reported if a trace is started by the SDK? I only see the ProfileSpans KV reported as part of the rack instrumentation.

Reply:
good point, I'll add that to the future SDK